# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: tools/profiler/core/platform.cpp
# Commit: 3aa39fa769a2
# Full Hash: 3aa39fa769a237babc6d2a09c39a4de58e9e589b
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2020-10-22 21:51:59
# Description:
#   Bug 1672501 - Initialize scProfilerMainThreadId from profiler_init - r=gregtatum
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D94419
# ==============================================================================

diff -r 766437adb95f -r 3aa39fa769a2 tools/profiler/core/platform.cpp
--- a/tools/profiler/core/platform.cpp	Thu Oct 22 12:46:28 2020 +0000
+++ b/tools/profiler/core/platform.cpp	Thu Oct 22 12:55:13 2020 +0000
@@ -185,7 +185,9 @@
 
 LazyLogModule gProfilerLog("prof");
 
-const int scProfilerMainThreadId = profiler_current_thread_id();
+// Statically initialized to 0, then set once from profiler_init(), which should
+// be called from the main thread before any other use of the profiler.
+int scProfilerMainThreadId;
 
 #if defined(GP_OS_android)
 class GeckoJavaSampler
@@ -3973,6 +3975,8 @@
 void profiler_init(void* aStackTop) {
   LOG("profiler_init");
 
+  scProfilerMainThreadId = profiler_current_thread_id();
+
   VTUNE_INIT();
 
   MOZ_RELEASE_ASSERT(!CorePS::Exists());